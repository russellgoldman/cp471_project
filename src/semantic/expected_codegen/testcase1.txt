num = 1
if num <= 10 goto L2
goto L3
L2: num = num + 1
L3: