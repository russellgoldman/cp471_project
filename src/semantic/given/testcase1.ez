Number num = 1;
while(num <= 10){
    num++;
}