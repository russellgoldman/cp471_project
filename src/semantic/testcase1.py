import unittest

class TestCase1(unittest.TestCase):
    def test_while_loop(self) -> None:
        source_f = open('./given/testcase1.ez', 'r')
        source = source_f.read()
        source_f.close()

        expected_f = open('./expected/testcase1.txt', 'r')
        expected = expected_f.read()
        expected_f.close()

if __name__ == '__main__':
    unittest.main()