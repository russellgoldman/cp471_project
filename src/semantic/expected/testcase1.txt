('statement', ('assignmentExpression', '=', 'num', 1), ('whileStatement', ('relationExpression', '<=', 'num', 10), ('assignmentExpression', 'num', '=', ('iterationExpression', '+', 'num', 1))))